src/sdk/endpoint_stats_collector.js

```diff
@@ -1,258 +1,289 @@
+    /**
+     * @param {EndpointStats} data 
+     * @returns {Promise<void>}
+     */
+    async _process_endpoint_stats(data) {
+
+        const bucket_counters = Object.entries(data.bucket_counters ?? {}).flatMap(
+            ([bucket_name, bkt]) => Object.entries(bkt).map(
+                ([content_type, io_stats]) => {
+                    dbg.log0(`bucket stats - ${bucket_name} ${content_type} :`, io_stats);
+                    return {
+                        bucket_name,
+                        content_type,
+                        read_count: io_stats.read_count,
+                        write_count: io_stats.write_count,
+                    };
+                }
+            )
+        );
+
+        const namespace_stats = Object.entries(data.namespace_stats ?? {}).map(
+            ([namespace_resource_id, io_stats]) => {
+                dbg.log0(`namespace stats - ${namespace_resource_id} :`, io_stats);
+                return {
+                    namespace_resource_id,
+                    io_stats: { ...io_stats }, // make shallow copy
+                };
+            }
+        );
 
-    async _send_endpoint_stats() {
-        await P.delay_unblocking(SEND_STATS_DELAY);
-        // clear this.send_stats to allow new updates to trigger another _send_endpoint_stats
-        this.send_stats = null;
-        try {
-            await this.rpc_client.object.update_endpoint_stats(this.get_all_stats(), {
+        if (this.rpc_client) {
+            await this.rpc_client.object.update_endpoint_stats({
+                bucket_counters,
+                namespace_stats,
+            }, {
                 timeout: SEND_STATS_TIMEOUT
             });
-            this.reset_all_stats();
-        } catch (err) {
-            // if update fails trigger _send_endpoint_stats again
-            dbg.error('failed on update_endpoint_stats. trigger _send_endpoint_stats again', err);
-            this._trigger_send_endpoint_stats();
         }
     }
 
-    async _send_nsfs_stats(attempts) {
-        await P.delay_unblocking(SEND_NSFS_STATS_DELAY);
-        const _nsfs_stats = {
-            nsfs_stats: {
-                io_stats: this.nsfs_io_counters,
-                op_stats: this.op_stats,
-                fs_workers_stats: this.fs_workers_stats,
-            }
-        };
-        // clear this.send_nsfs_stats to allow new updates to trigger another _send_nsfs_stats
-        this.send_nsfs_stats = null;
-        try {
-            await this.rpc_client.stats.update_nsfs_stats(_nsfs_stats, {
+    /**
+     * @param {NsfsStats} data
+     * @returns {Promise<void>}
+     */
+    async _process_nsfs_stats(data) {
+        dbg.log0('nsfs stats - IO counters :', data.io_stats);
+        for (const [k, v] of Object.entries(data.op_stats ?? {})) {
+            dbg.log0(`nsfs stats - S3 op=${k} :`, v);
+        }
+        for (const [k, v] of Object.entries(data.fs_workers_stats ?? {})) {
+            dbg.log0(`nsfs stats - FS op=${k} :`, v);
```

Comment:
Do we want them to be always printed? 
Maybe we want it to be printed only on a higher debug level?

WDYT?

Reply:
notice that only existing entries will print, so if there is no activity it is quiet.
I think this is useful - we don't log on every operation, so at least print periodic stats.